### PR TITLE
Create SQuAD metric README.md

### PR DESCRIPTION
**Pull request by sashavor**
Proposal for a metrics card structure (with an example based on the SQuAD metric).
@thomwolf @lhoestq @douwekiela @lewtun -- feel free to comment on structure or content (it's an initial draft, so I realize there's stuff missing!).

### CODE REVIEW

File: metrics/squad/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+The [original SQuAD paper](https://nlp.stanford.edu/pubs/rajpurkar2016squad.pdf) reported an F1 score of 51.0% and an Exact Match score of 40.0%. They also report that human performance on the dataset represents an F1 score of 90.5% and an Exact Match score of 80.3%.
+
+For more recent model performance, see the [dataset leaderboard](https://paperswithcode.com/dataset/squad).
+
+
+## Limitations and bias
+The SQuAD dataset does contain a certain amount of noise, such as duplicate questions as well as missing answers, but these represent a minority of the 100,000 question-answer pairs. Also, neither exact match nor F1 score reflect whether models do better on certain types of questions (e.g. who questions) or those that cover a certain gender or geographical area -- carrying out more in-depth error analysis can complement these numbers. 
+
+
+## Citation
```
Comment (thomwolf) [inline metrics/squad/README.md:37]: Maybe I would add a section with links to further references (blog post discussing the metrics, page in the HuggingFace course using the metric, etc)

**Review by thomwolf — COMMENTED** [review]
Very cool!

File: metrics/squad/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+## Metric description
```
Comment (lhoestq) [inline metrics/squad/README.md:1]: (just a title suggestion)
```suggestion
# Metric Card for SQuAD

## Metric description
```

File: metrics/squad/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    results = squad_metric.compute(predictions=predictions, references=references)
+  
+It outputs a dictionary with two values: the average exact match score and the average [F1 score](https://huggingface.co/metrics/f1) - see the [range section below](#range)
+
+    {'exact_match': 100.0, 'f1': 100.0}
+
+## Range
+The range of `exact_match` is 0-100, where 0.0 means no answers were matched and 100.0 means all answers were matched. 
+
+The range of `f1` is 0-1 -- its lowest possible value is 0, if either the precision or the recall is 0, and its highest possible value is 1.0, which means perfect precision and recall.
```
Comment (lhoestq) [inline metrics/squad/README.md:21]: Maybe this can be grouped in a "output values" section ?

**Review by lhoestq — COMMENTED** [review]
Cool thank you ! I also imagine that we could have a widget for people to play with the metric on the website somehow

File: metrics/squad/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## Values from popular papers
+The [original SQuAD paper](https://nlp.stanford.edu/pubs/rajpurkar2016squad.pdf) reported an F1 score of 51.0% and an Exact Match score of 40.0%. They also report that human performance on the dataset represents an F1 score of 90.5% and an Exact Match score of 80.3%.
+
+For more recent model performance, see the [dataset leaderboard](https://paperswithcode.com/dataset/squad).
+
+## Examples
```
[the remark this answers is not in the document]
Comment (sashavor) [reply metrics/squad/README.md:32]: We would need to update those regularly, no? that's why I prefer linking to an external resource..

[the remark this answers is not in the document]
Comment (sashavor) [reply metrics/squad/README.md:32]: Given that there are two outputs here, maybe that would be confusing? I think the code is clearer, somehow..

File: metrics/squad/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```python
+from datasets import load_metric
+squad_metric = load_metric("squad")
+results = squad_metric.compute(predictions=predictions, references=references)
+```
+## Output values
+
+This metric outputs a dictionary with two values: the average exact match score and the average [F1 score](https://huggingface.co/metrics/f1).
+
+    {'exact_match': 100.0, 'f1': 100.0}
```
Comment (lewtun) [inline metrics/squad/README.md:21]: Nit: maybe it's good to have all the code snippets and outputs as code blocks?

````suggestion
```
{'exact_match': 100.0, 'f1': 100.0}
```
````

File: metrics/squad/README.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    from datasets import load_metric
+    squad_metric = load_metric("squad")
+    predictions = [{'prediction_text': '1976', 'id': '56e10a3be3433e1400422b22'}, {'prediction_text': 'Beyonce', 'id': '56d2051ce7d4791d0090260b'},  {'prediction_text': 'climate change', 'id': '5733b5344776f419006610e1'}]
+    references = [{'answers': {'answer_start': [97], 'text': ['1976']}, 'id': '56e10a3be3433e1400422b22'}, {'answers': {'answer_start': [233], 'text': ['Beyoncé and Bruno Mars']}, 'id': '56d2051ce7d4791d0090260b'}, {'answers': {'answer_start': [891], 'text': ['climate change']}, 'id': '5733b5344776f419006610e1'}]
+    results = squad_metric.compute(predictions=predictions, references=references)
+    results
+    {'exact_match': 66.66666666666667, 'f1': 66.66666666666667}
+
+## Limitations and bias
+This metric works only with the [SQuAD v.1 dataset](https://huggingface.co/datasets/squad) -- it will not work with any other dataset formats.
```
Comment (lewtun) [inline metrics/squad/README.md:67]: Isn't it more accurate to say that this metric only works for datasets that have the same schema / format as SQuAD v1?

**Review by lewtun — COMMENTED** [review]
This is very cool and will really help people get a better understanding for how our metrics work!

I've left some tiny comments, but otherwise this looks great to me :)

File: metrics/squad/README.md (hunk 6 above)
Comment (sashavor) [reply metrics/squad/README.md:67]: well, it's technically a dataset-specific metric, cause there are ids that are internal to SQuAD, but you're right that they don't get checked, really.

File: metrics/squad/README.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+
+## Citation
+
+    @inproceedings{Rajpurkar2016SQuAD10,
+    title={SQuAD: 100, 000+ Questions for Machine Comprehension of Text},
+    author={Pranav Rajpurkar and Jian Zhang and Konstantin Lopyrev and Percy Liang},
+    booktitle={EMNLP},
+    year={2016}
+    }
```
Comment (lhoestq) [inline metrics/squad/README.md:85]:
````suggestion
```bibtex
@inproceedings{Rajpurkar2016SQuAD10,
title={SQuAD: 100, 000+ Questions for Machine Comprehension of Text},
author={Pranav Rajpurkar and Jian Zhang and Konstantin Lopyrev and Percy Liang},
booktitle={EMNLP},
year={2016}
}
```
````

**Review by lhoestq — APPROVED** [review]
Feel free to merge if it's all good for you :)

**Pull request merged**